pkg/docker/docker.go

```diff
@@ -126,6 +126,62 @@ type RunContainerOptions struct {
 	CapDrop          []string
 }
 
+// asDockerConfig converts a RunContainerOptions into a Config understood by the
+// docker client library.
+func (rco RunContainerOptions) asDockerConfig() docker.Config {
+	return docker.Config{
+		Image: getImageName(rco.Image),
+		User:  rco.User,
+		// TODO why do we call runContainerTar with a "partial"
+		// docker.Config?
+
+		Env: rco.Env,
+
+		// TODO review how callers use these options to pass stdin.
+		// Probably they are not necessary, as they're related to
+		// attaching to the container (that is often unnecessary).
```

Comment:
> (so we can see the logs)

This part is what I have shared previously when I implemented `runDocker` in origin, we don't need to attach anything to stream logs. It's much simpler to have a single goroutine calling the log api:
https://github.com/openshift/origin/blob/000489d894c9b54b749fbf9246d4d218197a7280/pkg/build/builder/dockerutil.go#L146-L193

> presumably we'd want to attach stderr when running save-artifacts in case the save-artifacts script uses it.

Again, we can stream stderr using the logs api.



> yeah so basically we need to attach stdin for assemble (to stream in the content).

I believe we don't need to "attach" stdin to send content, it should be enough to set the Stdin field when creating the container. Needs confirmation.